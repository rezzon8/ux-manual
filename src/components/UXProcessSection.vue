<template>
  <v-container>
    <v-row>
      <v-col cols="12">
        <h4 class="display-1">{{ title }}</h4>
      </v-col>
      <v-col
        v-for="item in process_steps"
        :key="item.id"
        :href="item.href"
        class="mb-5 d-flex align-stretch"
        cols="4"
      >
        <v-card
          outlined
          v-for="(copy, index) in item.copy.introduction"
          :key="index"
          class="d-flex align-start flex-column"
        >
          <v-card-title class="primary mb-2" style="width: 100%">
            {{ item.title }}
          </v-card-title>
          <v-card-text>
            {{ copy }}
          </v-card-text>
          <v-card-actions class="mt-auto align-self-end">
            <v-btn outlined depressed color="primary white--text">
              View
            </v-btn>
          </v-card-actions>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
export default {
  name: "UXProcessSection",

  props: {
    process_steps: Array,
    title: String
  },

  data: () => ({}),
  computed: {}
};
</script>
