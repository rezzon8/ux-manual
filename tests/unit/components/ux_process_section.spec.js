import { mount, createLocalVue } from "@vue/test-utils";
import Vue from "vue";
import Vuetify from "vuetify";
import UXProcessSection from "@/components/UXProcessSection.vue";
Vue.use(Vuetify); // added to prevent vuetify lint errors
const localVue = createLocalVue();
const props = {
  ux_processes: [
    {
      id: 1,
      title: "UX Audit",
      copy:
        "After a kickoff meeting about your goals and mission, we conduct a UX review of your current status. As a result, you get an analysis stating the opportunities to develop. We offer UX Audit as an independent service as well.",
      color: "#3f51b5"
    }
  ],
  process_steps: [
    {
      id: 1,
      ux_process_id: 1,
      title: "Stakeholder Interview",
      copy: {
        introduction: [
          "We kick-off each project with a meeting with you: we need to talk about your mission, goals and context. This meeting ensures that we start the UX design process by understanding the problem. We also need to have a good knowledge of your domain in order to ask the right questions, find the best solutions."
        ]
      }
    }
  ]
};
describe("UXProcessSection.vue", () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });
  it("should have a custom title", () => {
    const props = {
      title: "new message"
    };
    const wrapper = mount(UXProcessSection, {
      propsData: { title: props.title }
    });
    expect(wrapper.text()).toMatch(props.title);
  });
  it("should render a list of process_steps", () => {
    const wrapper = mount(UXProcessSection, {
      localVue,
      vuetify,
      propsData: { process_steps: props.process_steps }
    });
    expect(wrapper.find(".v-card__title").text()).toMatch(
      "Stakeholder Interview"
    );
  });
  it("should emit a click handler", async () => {
    const wrapper = mount(UXProcessSection, {
      localVue,
      vuetify,
      propsData: { process_steps: props.process_steps }
    });
    console.log(wrapper.html());
    await wrapper.trigger("click");
    console.log(wrapper.html());
  });
});
