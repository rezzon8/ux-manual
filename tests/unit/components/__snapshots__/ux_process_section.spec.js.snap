// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UXProcessSection.vue should have a custom title 1`] = `
<div class="container">
  <div class="row">
    <div class="col col-12">
      <h4 class="display-1">new message</h4>
    </div>
  </div>
</div>
`;

exports[`UXProcessSection.vue should render a list of process_steps 1`] = `
<div class="container">
  <div class="row">
    <div class="col col-12">
      <h4 class="display-1"></h4>
    </div>
    <div class="mb-5 d-flex align-stretch col col-4">
      <div class="d-flex align-start flex-column v-card v-sheet v-sheet--outlined theme--light">
        <div class="v-card__title primary mb-2" style="width: 100%;">
          Stakeholder Interview
        </div>
        <div class="v-card__text">
          We kick-off each project with a meeting with you: we need to talk about your mission, goals and context. This meeting ensures that we start the UX design process by understanding the problem. We also need to have a good knowledge of your domain in order to ask the right questions, find the best solutions.
        </div>
        <div class="v-card__actions mt-auto align-self-end"><button type="button" class="v-btn v-btn--depressed v-btn--flat v-btn--has-bg v-btn--outlined theme--light v-size--default primary white--text"><span class="v-btn__content">
            View
          </span></button></div>
      </div>
    </div>
  </div>
</div>
`;
